messaging/src/main/java/org/axonframework/common/infra/ComponentDescriptor.java

```diff
@@ -16,24 +16,103 @@
 
 package org.axonframework.common.infra;
 
+import jakarta.annotation.Nonnull;
+
 import java.util.Collection;
 import java.util.Map;
 
+/**
+ * Contract towards describing the properties an (infrastructure) component might have.
+ * <p>
+ * The {@link ComponentDescriptor} implementation passed to a component by
+ * {@link DescribableComponent#describeTo(ComponentDescriptor)} determines both the structure and
+ * format of the output. This means the same component can be rendered in different formats (like JSON, XML, or a
+ * filesystem-like tree) depending on which {@link ComponentDescriptor} implementation is used. The component only needs
+ * to provide its properties and structure to the descriptor, which then handles the formatting details.
+ * <p>
+ * <strong>Handling Circular References:</strong> Component hierarchies may contain circular references where a
+ * {@link DescribableComponent} refers to another component that eventually references back to the original.
+ * {@link ComponentDescriptor} implementations must handle these circular dependencies to prevent infinite recursion.
+ *
+ * @author Allard Buijze
```

Comment:
I think your name deserves to be here too, @MateuszNaKodach  :-)

Reply:
As you wish :)